test/collection_test.js

```diff
@@ -456,6 +471,16 @@ describe("Collection", () => {
         .should.eventually.be.a("number");
     });
 
+    it("should accept a record for the 'generate' function", () => {
+      articles = testCollection({
+        idSchema: createKeyListIdSchema()
+      });
+
+      return articles.create(article)
+        .then(result => result.data.id.split(","))
+        .should.eventually.be.a("Array");
```

Comment:
I'd rather directly check for the resulting string. Though I suspect you did this to avoid having intermittent failures because object keys order is never guaranteed; if so, we should probably ensure always sorting the keys before joining.